Reduce the margin of the block toolbar.

Alex-Kostov opened the pull request:
Reduced the margin of the block toolbar
Fixes #38391

<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

## Description
Reduced the spacing above the toolbar, now it sits evenly.

## Testing Instructions
1. Activate a classic theme.
2. Go to Appearance → Customize → Widgets.
3. Select the first block.
4. Check if the spacing on the top is reduced.

## Screenshots <!-- if applicable -->
![](https://i.imgur.com/AyNgMJQ.png)

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [x] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

Alex-Kostov committed:
Fix for widget customizer panel,
Reduced the margin of the block toolbar
Fixes #38391

noisysocks commented:
Thanks for fixing this up, @Alex-Kostov!

Here's how it looks now:

<img width="364" alt="Screen Shot 2022-02-14 at 15 01 18" src="https://user-images.githubusercontent.com/612155/153798234-6c5b151e-a51e-4049-9b69-2fad816f261c.png">

I believe in the original design there was a little bit of gap above the toolbar. Changing `grid-margin-10` to `grid-margin-20` looks about right to me:

<img width="343" alt="Screen Shot 2022-02-14 at 15 06 20" src="https://user-images.githubusercontent.com/612155/153798433-f8a7e1f3-afca-44e6-ac08-fd6d7df35302.png">

I'll make that change for you since it's a quick one.

Hope to see you around the Gutenberg repo now that you know how to make changes and submit a PR! Lots more issues with a `Good First Issue` label if you're interested 🙂

noisysocks commented on packages/customize-widgets/src/components/header/style.scss:
```diff
@@ -1,7 +1,7 @@
 	@include break-medium() {
 		// Make space for the floating toolbar.
-		margin-bottom: $grid-unit-60 + $default-block-margin;
+		margin-bottom: $grid-unit-10 + $default-block-margin;
```
```suggestion
		margin-bottom: $grid-unit-20 + $default-block-margin;
```

noisysocks committed:
Increase padding a little

noisysocks commented:
The failing E2E test here is because of a pre-existing bug in `trunk` that's been unsurfaced by this change. It seems that the block toolbar doesn't get dismissed when opening the slide-out inserter which means that it obscures one of the inserter buttons that Puppeteer tries to click. Investigating a fix...

noisysocks committed:
Dismiss block toolbar after opening inserter in customizing-widgets E2E tests

The block toolbar sometimes obscures buttons in the inserter causing
test failures. It's not clear whether this is desired behaviour or not.
For now, let's just work around it in the E2E test.

noisysocks commented:
I worked around the test failure for now. Thanks for the patch @Alex-Kostov!
